File: specification/specs/search/search/hits/Hit.ts
class Hit<TDocument> {
  _index: IndexName;
  _id: Id;

  _score?: double;
  _type?: TypeName;

  _explanation?: Explanation;
  fields?: Dictionary<string, LazyDocument>;
  highlight?: Dictionary<string, string[]>;
  /** @prop_serializer VerbatimInterfaceReadOnlyDictionaryKeysFormatter`2 */
  inner_hits?: Dictionary<string, InnerHitsResult>;
  matched_queries?: string[];
  _nested?: NestedIdentity;
  _ignored?: string[];

  _shard?: string;
  _node?: string;
  _routing?: string;
  _source?: TDocument;
  _seq_no?: long;
  _primary_term?: long;
  _version?: long;
  sort?: Array<long | double | string>;
}

File: output/csharp/Search/Search/Hits/Hit.cs
using Nest.Search;
using Nest.CommonAbstractions;
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Search {

	public class Hit<TDocument>  {
		
		[DataMember(Name="_explanation")]
		public Explanation Explanation { get; set; }


		[DataMember(Name="fields")]
		public IDictionary<string, LazyDocument> Fields { get; set; }


		[DataMember(Name="highlight")]
		public IDictionary<string, List<string>> Highlight { get; set; }


		[DataMember(Name="inner_hits")]
		public IDictionary<string, InnerHitsResult> InnerHits { get; set; }


		[DataMember(Name="matched_queries")]
		public List<string> MatchedQueries { get; set; }


		[DataMember(Name="_nested")]
		public NestedIdentity Nested { get; set; }


		[DataMember(Name="_score")]
		public double Score { get; set; }


		[DataMember(Name="sort")]
		public List<object> Sort { get; set; }

	}
}
